app/controllers/users/reactivate_profile_controller.rb

```diff
@@ -19,7 +19,12 @@ def create
     def build_reactivate_profile_form
       ReactivateProfileForm.new(
         current_user,
-        params[:reactivate_profile_form].permit(:recovery_code, :password)
+        params[:reactivate_profile_form].
+        permit({:recovery_code => []}, :password).
+        tap do |obj|
+          code = obj[:recovery_code]
+          obj[:recovery_code] = code.join(' ') unless obj[:recovery_code].nil?
```

Comment:
Also, I just made a test case, I believe that the syntax `permit(:password, recovery_code: [])` should definitely work:

```ruby
params = ActionController::Parameters.new(reactivate_profile_form: { password: 'foobar', recovery_code: %w(a b c d) })
=> {"reactivate_profile_form"=>{"password"=>"foobar", "recovery_code"=>["a", "b", "c", "d"]}}
params[:reactivate_profile_form].permit(:password, recovery_code: [])
=> {"password"=>"foobar", "recovery_code"=>["a", "b", "c", "d"]}
irb(main):013:0> 
```